libraries/chain/include/eosio/chain/controller.hpp

```diff
@@ -328,9 +328,9 @@ namespace eosio { namespace chain {
 
          signal<void(uint32_t)>                        block_start; // block_num
          signal<void(const signed_block_ptr&)>         pre_accepted_block;
-         signal<void(const block_state_ptr&)>          accepted_block_header;
-         signal<void(const block_state_ptr&)>          accepted_block;
-         signal<void(const block_state_ptr&)>          irreversible_block;
+         signal<void(const block_state_legacy_ptr&)>   accepted_block_header;
+         signal<void(const block_state_legacy_ptr&)>   accepted_block;
+         signal<void(const block_state_legacy_ptr&)>   irreversible_block;
```

Comment:
Should not be part of this PR.

Reply:
Although, third-party plugins would need to be updated after this is merged. No third-party plugins will be impacted until 6.0 release. Before then, this will have to change one way or another anyway.

Reply:
Thanks. I will go ahead merging this into main in 2 hours.